lib/matplotlib/tri/tricontour.py

```diff
@@ -79,12 +79,22 @@ def _contour_args(self, args, kwargs):
             fn = 'contour'
         tri, args, kwargs = Triangulation.get_from_args_and_kwargs(*args,
                                                                    **kwargs)
-        z = np.asarray(args[0])
+        z = np.ma.asarray(args[0])
         if z.shape != tri.x.shape:
             raise ValueError('z array must have same length as triangulation x'
                              ' and y arrays')
-        self.zmax = z.max()
-        self.zmin = z.min()
+
+        # z values must be finite, only need to check points that are included
+        # in the triangulation.
+        z_check = z[np.unique(tri.get_masked_triangles())]
+        if np.ma.is_masked(z_check):
+            raise ValueError('z cannot be a masked array')
+        if not np.isfinite(z_check).all():
+            raise ValueError('z array cannot contain non-finite values')
```

Comment:
Similarly, maybe "z array must not contain non-finite values within the triangulation".